python/pyarrow/_parquet.pyx

```diff
@@ -1334,6 +1339,15 @@ cdef class ParquetWriter:
             for column in self.use_dictionary:
                 props.enable_dictionary(column)
 
+    cdef void _set_byte_stream_split_props(
+            self, WriterProperties.Builder* props):
+        if isinstance(self.use_byte_stream_split, bool):
+            if self.use_byte_stream_split:
+                props.encoding(ParquetEncoding_BYTE_STREAM_SPLIT)
```

Comment:
Will that enable byte_stream_split selectively for floating-point columns? Or will that raise an error if there are non-floating-point columns?

Reply:
The current behavior is:
If dictionary is used, then no exception is raised. The error gets generated when an attempt is made to create an byte_stream_split encoder for non-floating-point types.

I extended my smoke test to check that an exception is thrown and propagated to pyarrow if integer data is attempted to be encoded using that encoding.